include/onnxruntime/core/framework/data_types.h

```diff
@@ -694,6 +700,59 @@ class NonOnnxType : public DataTypeImpl {
   NonOnnxType() = default;
 };
 
+class PrimitiveDataTypeBase : public DataTypeImpl {
+ public:
+  bool IsCompatible(const ONNX_NAMESPACE::TypeProto&) const override {
+    return false;
+  }
+
+  const PrimitiveDataTypeBase* AsPrimitiveDataType() const override final {
+    return this;
+  }
+
+  const ONNX_NAMESPACE::TypeProto* GetTypeProto() const final {
+    return nullptr;
+  }
+
+  int GetTensorElementType() const {
+    return data_type_;
+  }
+
+ protected:
+  PrimitiveDataTypeBase() = default;
+
+  void set_data_type(int32_t data_type) {
+    data_type_ = data_type;
+  }
+
+ private:
+  int32_t data_type_;
```

Comment:
Any reason you store it as int32_t and return as int from GetTensorElementType?

Reply:
I saw some other place in the code that use int32_t for the purpose and it made sense to me. I don't have to deal with different sizes of int on different platforms.

---
In reply to: [341771388](https://github.com/microsoft/onnxruntime/pull/2307#discussion_r341771388) [](ancestors = 341771388)

Reply:
This comment was not for the usage of int32_t, but the usage of int as the return type of GetElementType. I see that you've already fixed that now.